pkg/reporter/json_report.go

```diff
@@ -0,0 +1,184 @@
+package reporter
+
+import (
+	"fmt"
+	"os"
+	"encoding/json"
+	"github.com/safedep/vet/pkg/analyzer"
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/policy"
+	"github.com/safedep/vet/pkg/exceptions"
+)
+
+
+type JsonReportingConfig struct {
+	Path string
+}
+
+// Highlevel Json Structure
+type jsonTemplateInput struct {
+	Remediations       map[string][]jsonTemplateInputRemediation `json:"remediations"`         // Remediations
+	Summary            jsonTemplateInputResultSummary 			 `json:"summary"`              // Summary
+	Violations         []jsonTemplateInputViolation              `json:"violations"`           // Violations
+}
+
+
+type jsonTemplateInputViolation struct {
+	Ecosystem string `json:"ecosystem"` // Ecosystem of the package
+	PkgName   string `json:"pkg_name"`  // Package name
+	Message   string `json:"message"`   // Violation message
+}
+
+type jsonTemplateInputRemediation struct {
+	Pkg                *models.Package `json:"pkg"`                  // Package details
+	PkgRemediationName string          `json:"pkg_remediation_name"` // Remediation name
+	Score              int             `json:"score"`                // Score
+	Tags               string          `json:"tags"`                 // Tags
+}
+
+type jsonTemplateInputManifestResultSummary struct {
+	Ecosystem               string `json:"ecosystem"`                 // Ecosystem
+	PackageCount            int    `json:"package_count"`             // Package count
+	PackageWithIssuesCount  int    `json:"package_with_issues_count"` // Packages with issues count
+}
+
+type jsonTemplateInputResultSummary struct {
```

Comment:
This JSON report schema probably needs a spec. I will go ahead and merge this but we need to enhance this so that we can offer the JSON report schema as a contract defined outside `vet`, ideally in a proto spec file and generate the Go structs required for building the report.

The reason for doing this is:

1. Users may leverage the JSON schema for their integrations
2. We need to build an extensible & evolvable schema that serves as a contract for JSON reporting
3. JSON report generator module in `vet` should conform to this schema